### PR TITLE
Collecting React rendering errors

### PR DESCRIPTION
**Pull request by rtrieu**
### What does this PR do?
Add details about how to instrument the React router in order to monitor React rendering errors with RUM Browser SDK.

### Motivation
Request from @nhientrandatadog and contribution from @amortemousque, as per [DOCS-6097](https://datadoghq.atlassian.net/browse/DOCS-6097).

### Additional Notes
Ready to merge following a review/approval from @amortemousque 

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

[DOCS-6097]: https://datadoghq.atlassian.net/browse/DOCS-6097?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 1]
```diff
@@ -140,6 +140,88 @@ try {
 {{% /tab %}}
 {{< /tabs >}}
 
+### Collect React rendering errors
+
+Error boundaries are React components that catch rendering errors anywhere in its child component tree and display a fallback UI. React error boundaries offer the capability to collect the component stack. Compared to error stack traces, React error boundaries provide more detailed information about the hierarchy of components involved, which ease developer identifying and fixing errors.
```
Comment (urseberry) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:145]:
```suggestion
Error boundaries are React components that catch rendering errors anywhere in its child component tree and display a fallback UI. React error boundaries offer the capability to collect the component stack. Compared to error stack traces, React error boundaries provide more detailed information about the hierarchy of components involved. The additional information helps developers identify and fix errors.
```

**Review by urseberry — COMMENTED** [review]
💯

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 2]
```diff
@@ -140,6 +140,88 @@ try {
...
 
+### Collect React rendering errors
+
+Error boundaries are React components that catch rendering errors anywhere in its child component tree and display a fallback UI. React error boundaries offer the capability to collect the component stack. Compared to error stack traces, React error boundaries provide more detailed information about the hierarchy of components involved. The additional information helps developers identify and fix errors.
+
+If your React application throws an error during rendering, React removes its UI from the screen by default. To modify this behavior, you can instrument the React router to monitor React rendering errors with the RUM Browser SDK `addError()` API.
+
+The collected rendering error stack contains the component stack, which is unminified like any other error stack traces once you've uploaded sourcemaps.
+
+**Note**: These instructions are specific to the **React Router v6** library.
```
Comment (amortemousque) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:151]: There is no relation between React router and React Error boundaries
```suggestion
```

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 3]
```diff
@@ -140,6 +140,88 @@ try {
 {{% /tab %}}
 {{< /tabs >}}
 
+### Collect React rendering errors
+
+Error boundaries are React components that catch rendering errors anywhere in its child component tree and display a fallback UI. React error boundaries offer the capability to collect the component stack. Compared to error stack traces, React error boundaries provide more detailed information about the hierarchy of components involved. The additional information helps developers identify and fix errors.
+
+If your React application throws an error during rendering, React removes its UI from the screen by default. To modify this behavior, you can instrument the React router to monitor React rendering errors with the RUM Browser SDK `addError()` API.
+
+The collected rendering error stack contains the component stack, which is unminified like any other error stack traces once you've uploaded sourcemaps.
```
Comment (amortemousque) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:149]: People looking for a React integration are likely already familiar with Error boundaries. So IMO, there is no need to describe the behaviour of React Error boundaries. 
We could stay concise like for [React router integration](https://docs.datadoghq.com/real_user_monitoring/browser/modifying_data_and_context/?tab=npm#react-router-instrumentation) and only say:


```suggestion
You can instrument the React error boundaries to monitor React rendering errors using the RUM Browser SDK addError() API.
The collected rendering error stack contains the component stack, which is unminified like any other error stacktraces once you’ve uploaded sourcemaps.

To instrument React error boundaries for RUM monitoring, use the following:
```

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 4]
```diff
@@ -140,6 +140,88 @@ try {
 {{% /tab %}}
 {{< /tabs >}}
 
+### Collect React rendering errors
```
[the remark this answers is not in the document]
Comment (amortemousque) [reply content/en/real_user_monitoring/browser/collecting_browser_errors.md:144]: I think there was a mistake in the commits. The title is back to `Collect React rendering errors`

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md (hunk 2 above)
Comment (amortemousque) [reply content/en/real_user_monitoring/browser/collecting_browser_errors.md]: I think there was a mistake in the commits. The line is back

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 5]
```diff
@@ -140,6 +141,86 @@ try {
...
+
+You can instrument the React error boundaries to monitor React rendering errors using the RUM Browser SDK `addError()` API.
+
+The collected rendering error stack contains the component stack, which is unminified like any other error stack traces once you've uploaded sourcemaps.
```
Comment (amortemousque) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:148]: **🥜 nitpick:** ‏The collected rendering errors contain a component stack, which is unminified like any other error stack traces once you've uploaded sourcemaps.

**Comment by rtrieu** [comment]
@amortemousque Oops...I might have clicked the wrong version when resolving conflicts. Thanks so much for catching these!

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 6]
```diff
@@ -140,6 +141,84 @@ try {
 {{% /tab %}}
 {{< /tabs >}}
 
+### React error boundaries instrumentation 
+
+You can instrument the React error boundaries to monitor React rendering errors using the RUM Browser SDK `addError()` API.
```
Comment (bcaudan) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:146]: **💬 suggestion:** ‏what about adding a link to react error boundaries doc?

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 7]
```diff
@@ -140,6 +141,84 @@ try {
...
+
+The collected rendering errors contain a component stack, which is unminified like any other error stack traces once you've uploaded sourcemaps.
```
Comment (bcaudan) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:148]: **💬 suggestion:** ‏what about adding a link to sourcemaps upload doc?

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md (hunk 6 above)
Comment (rtrieu) [reply content/en/real_user_monitoring/browser/collecting_browser_errors.md]: Added, thanks for the suggestions!

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 8]
```diff
@@ -58,7 +59,7 @@ addError(
 );
 {{< /code-block >}}
 
-**Note**: The [Error Tracking][4] feature processes errors sent with source set to `custom` or `source` and that contain a stack trace. Errors sent with any other source (such as `console`) will not be processed by Error Tracking.
+**Note**: The [Error Tracking][4] feature processes errors sent with source set to `custom` or `source` and that contain a stack trace. Errors sent with any other source (such as `console`) are not be processed by Error Tracking.
```
Comment (urseberry) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:62]:
```suggestion
**Note**: The [Error Tracking][4] feature processes errors sent with source set to `custom` or `source` and that contain a stack trace. Errors sent with any other source (such as `console`) are not processed by Error Tracking.
```

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 9]
```diff
@@ -140,6 +141,84 @@ try {
 {{% /tab %}}
 {{< /tabs >}}
 
+### React error boundaries instrumentation 
+
+You can instrument the React [error boundaries][5] to monitor React rendering errors using the RUM Browser SDK `addError()` API.
+
+The collected rendering errors contain a component stack, which is unminified like any other error stack traces once you've [uploaded sourcemaps][6].
```
Comment (urseberry) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:148]:
```suggestion
The collected rendering errors contain a component stack, which is unminified like any other error stack traces after you [upload sourcemaps][6].
```
Vale made me do it. 😄

**Review by urseberry — APPROVED** [review]

**Pull request merged**